metals/src/main/scala/scala/meta/internal/metals/UserConfiguration.scala

```diff
@@ -1,44 +1,92 @@
 package scala.meta.internal.metals
 
 import com.google.gson.JsonObject
+import com.google.gson.JsonPrimitive
+import java.util.Properties
 import scala.collection.mutable.ListBuffer
-import scala.util.control.NonFatal
+import scala.util.Try
 
 /**
  * Configuration that the user can override via workspace/didChangeConfiguration.
  *
  * @param javaHome The Java home location used to detect src.zip for JDK sources.
  */
 case class UserConfiguration(
-    javaHome: Option[String] = None
+    javaHome: Option[String] = None,
+    sbtLauncher: Option[String] = None,
+    sbtOpts: List[String] = Nil
 )
 
 object UserConfiguration {
+  def options: List[UserConfigurationOption] = List(
+    UserConfigurationOption(
+      "java-home",
+      "`JAVA_HOME` environment variable with fallback to `user.home` system property.",
+      "/Library/Java/JavaVirtualMachines/jdk1.8.0_192.jdk/Contents/Home",
+      "Java Home directory",
+      "The Java Home directory used for indexing JDK sources and locating the `java` binary."
+    ),
+    UserConfigurationOption(
+      "sbt-launcher",
+      "Metals embedded sbt-launch.jar.",
+      "/usr/local/Cellar/sbt/1.2.6/libexec/bin/sbt-launch.jar",
```

Comment:
I'm not sure I understand, why is this hardcoded here?

Reply:
That's the `example` field ;)

 (took me a while too)